src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,78 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION = 0;
+    private static final int AMOUNT = 1;
+    private static final String REGEX = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder stringBuilder = readTheFile(fromFileName);
+        String resultData = createReport(stringBuilder);
+        writeTheFile(toFileName, resultData);
+    }
+
+    private StringBuilder readTheFile(String fromFileName) {
+        File file = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("can`t read the file" + fromFileName, e);
+        }
+        return stringBuilder;
```

Comment:
```suggestion
        return stringBuilder.toString();
```

Reply:
"Why I can`t use void methods. Is it forbidden or ... I dont understand"

It can be used if this method should simply do something and you do not expect any data from it.
The task of this method is to read the file. We can’t just read, we need data from there. So we read and return data from the file to use elsewhere.